### PR TITLE
Allow deleting a room

### PR DESCRIPTION
**Pull request by dmathieu**
```ruby
client['room_id'].delete_room
```

### CODE REVIEW

File: spec/hipchat_api_v2_spec.rb [hunk 1]
```diff
@@ -379,6 +379,15 @@
     end
   end
 
+  describe "#delete_room" do
```
Comment (rberrelleza) [inline spec/hipchat_api_v2_spec.rb]: Can you also add a test for when the room is not found? To verify the error is handled properly.

Comment (dmathieu) [inline spec/hipchat_api_v2_spec.rb]: Sure. I've done it for the delete test. There's no such test for any other endpoint and I haven't added them though.

Comment (rberrelleza) [inline spec/hipchat_api_v2_spec.rb]: (awthanks)

**Comment by dmathieu** [comment]
@rberrelleza I added support for deleting rooms with the V1 API and rebased to fix a conflict.

**Comment by rberrelleza** [comment]
Thanks for the contribution, this is great!

**Comment by dmathieu** [comment]
Thank you. :tulip:

**Pull request merged**